tests/model/test_model.py

```diff
@@ -0,0 +1,235 @@
+from math import nan, inf
+from pathlib import Path
+
+import networkx as nx
+import numpy as np
+import pandas as pd
+import pytest
+
+from cascade.model import (
+    Session, Model, DismodGroups, SmoothGrid, Var, Covariate,
+    Uniform, Gaussian
+)
+from cascade.stats.compartmental import siler_default, total_mortality_solution
+
+
+@pytest.fixture
+def basic_model():
+    nonzero_rates = ["iota", "chi", "omega"]
+    locations = nx.DiGraph()
+    locations.add_edges_from([(1, 2), (1, 3), (1, 4)])
+    eta = 1e-3
+
+    parent_location = 1
+    child_locations = list()
+    covariates = [Covariate("traffic", reference=0.0)]
+    model = Model(nonzero_rates, parent_location, child_locations, covariates=covariates)
+
+    covariate_age_time = ([40], [2000])
+    traffic = SmoothGrid(covariate_age_time)
+    traffic.value[:, :] = Gaussian(lower=-1, upper=1, mean=0.00, standard_deviation=0.3, eta=eta)
+    model.alpha[("traffic", "iota")] = traffic
+
+    dense_age_time = (np.linspace(0, 120, 13), np.linspace(1990, 2015, 8))
+    rate_grid = SmoothGrid(dense_age_time)
+    rate_grid.value[:, :] = Uniform(lower=1e-6, upper=0.3, mean=0.001, eta=eta)
+    rate_grid.dage[:, :] = Uniform(lower=-1, upper=1, mean=0.0, eta=eta)
+    rate_grid.dtime[:, :] = Gaussian(lower=-1, upper=1, mean=0.0, standard_deviation=0.3, eta=eta)
+
+    model.rate["omega"] = rate_grid
+    model.rate["iota"] = rate_grid
+
+    chi_grid = SmoothGrid(dense_age_time)
+    chi_grid.value[:, :] = Uniform(lower=1e-6, upper=0.3, mean=0.004, eta=eta)
+    chi_grid.dage[:, :] = Uniform(lower=-.9, upper=.9, mean=0.0, eta=eta)
+    chi_grid.dtime[:, :] = Gaussian(lower=-.8, upper=.8, mean=0.0, standard_deviation=0.4, eta=eta)
+    model.rate["chi"] = chi_grid
+    return model
+
+
+def test_write_rate(basic_model, dismod):
+    locations = pd.DataFrame(dict(
+        name=["global"],
+        parent=[nan],
+        c_location_id=[1],
+    ))
+    parent_location = 1
+    db_file = Path("rftest.db")
+    session = Session(locations, parent_location, db_file)
+
+    # This peeks inside the session to test some of its underlying functionality
+    # without doing a fit.
+    session.write_model(basic_model, ([], []))
+    session._run_dismod(["init"])
```

Comment:
Doesn't session.write_model already run init?

Reply:
No, it doesn't. We should be able to test `write_model` against an in-memory database, at some point. It does call flush() but not Dismod-AT.

Reply:
Ok